include/volume/volume_cooling_gaussians_crhmc.hpp

```diff
@@ -0,0 +1,448 @@
+#ifndef VOLUME_COOLING_GAUSSIANS_CRHMC_HPP
+#define VOLUME_COOLING_GAUSSIANS_CRHMC_HPP
+
+//#define VOLESTI_DEBUG
+
+#include "volume/volume_cooling_gaussians.hpp"
+#include "preprocess/crhmc/crhmc_problem.h"
+
+////////////////////////////// Algorithms
+
+// Gaussian Anealling
+
+// Implementation is based on algorithm from paper "A practical volume algorithm",
+// Springer-Verlag Berlin Heidelberg and The Mathematical Programming Society 2015
+// Ben Cousins, Santosh Vempala
+
+// Compute a_{i+1} when a_i is given
+template
+<
+    typename WalkType,
```

Comment:
since this is specialized for crhmc you can avoid creating a lot of templates, e.g. WalkType, RandomPointGenerator, CrhmcProblem etc. Please make the code specific for that walk no need to be generic